x/evm/types/ethtx/access_list_tx.go

```diff
@@ -0,0 +1,205 @@
+package ethtx
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+func NewAccessListTx(tx *ethtypes.Transaction) (altx *AccessListTx, reserr error) {
+	txData := &AccessListTx{
+		Nonce:    tx.Nonce(),
+		Data:     tx.Data(),
+		GasLimit: tx.Gas(),
+	}
+
+	v, r, s := tx.RawSignatureValues()
+
+	defer func() {
+		if err := recover(); err != nil {
+			altx = nil
+			reserr = fmt.Errorf("%s", err)
+		}
+	}()
```

Comment:
What's this for? Why do we need this?

Reply:
yeah it's to capture panics that might be thrown by `MustSetConvertIfPresent`. There might be a way to handle it without involving panic

Reply:
updated